src/main/kotlin/no/nav/k9/nyoppgavestyring/domeneadaptere/k9saktillos/K9SakTilLosAdapterTjeneste.kt

```diff
@@ -0,0 +1,248 @@
+package no.nav.k9.nyoppgavestyring.domeneadaptere.k9saktillos
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import no.nav.k9.domene.repository.BehandlingProsessEventK9Repository
+import no.nav.k9.integrasjon.kafka.dto.BehandlingProsessEventDto
+import no.nav.k9.kodeverk.behandling.BehandlingStatus
+import no.nav.k9.nyoppgavestyring.mottak.feltdefinisjon.FeltdefinisjonTjeneste
+import no.nav.k9.nyoppgavestyring.mottak.feltdefinisjon.Feltdefinisjoner
+import no.nav.k9.nyoppgavestyring.mottak.feltdefinisjon.FeltdefinisjonerDto
+import no.nav.k9.nyoppgavestyring.mottak.omraade.OmrådeRepository
+import no.nav.k9.nyoppgavestyring.mottak.oppgave.OppgaveDto
+import no.nav.k9.nyoppgavestyring.mottak.oppgave.OppgaveFeltverdiDto
+import no.nav.k9.nyoppgavestyring.mottak.oppgave.OppgaveV3Tjeneste
+import no.nav.k9.nyoppgavestyring.mottak.oppgavetype.OppgavetypeTjeneste
+import no.nav.k9.nyoppgavestyring.mottak.oppgavetype.OppgavetyperDto
+import no.nav.k9.sak.kontrakt.aksjonspunkt.AksjonspunktTilstandDto
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.util.concurrent.TimeUnit
+import kotlin.concurrent.fixedRateTimer
+import no.nav.k9.Configuration
+
+class K9SakTilLosAdapterTjeneste(
+    private val behandlingProsessEventK9Repository: BehandlingProsessEventK9Repository,
+    private val områdeRepository: OmrådeRepository,
+    private val feltdefinisjonTjeneste: FeltdefinisjonTjeneste,
+    private val oppgavetypeTjeneste: OppgavetypeTjeneste,
+    private val oppgaveV3Tjeneste: OppgaveV3Tjeneste,
+    private val config: Configuration
+) {
+
+    private val log: Logger = LoggerFactory.getLogger(K9SakTilLosAdapterTjeneste::class.java)
+
+    fun kjør(kjørSetup: Boolean) {
+        if (config.nyOppgavestyringAktivert()) {
+            fixedRateTimer(
+                name = "k9-sak-til-los",
+                daemon = true,
+                initialDelay = TimeUnit.SECONDS.toMillis(10),
+                period = TimeUnit.DAYS.toMillis(1)
+            ) {
+                if (kjørSetup) {
+                    setup()
+                }
+                spillAvBehandlingProsessEventer()
+            }
+        }
+    }
+
+    fun spillAvBehandlingProsessEventer() {
+        log.info("Starter avspilling av BehandlingProsessEventer")
+        behandlingProsessEventK9Repository.hentAlleEventerIder()
+            .map { uuid ->
+                behandlingProsessEventK9Repository.hent(uuid).eventer
+                    .map { event ->
+                        event.aksjonspunktTilstander.forEach { aksjonspunktTilstand ->
+                            var oppgaveDto = lagOppgaveDto(event, aksjonspunktTilstand)
```

Comment:
Jeg tror `status` på OppgaveDto kan bli feil når det er flere aksjonspunkter med ulik status på samme event. 
Kun status på den første aksjonspunkttilstanden blir brukt, deretter ignoreres resten pga duplikatsjekk på event.eksternId og event.eventTid

Reply:
Så nå at mappingen av aksjonspunktene egentlig gjøres i `lagFeltverdier()`.
Denne loopen er i såfall litt overflødig: `event.aksjonspunktTilstander.forEach { aksjonspunktTilstand ->`

Reply:
Litt usikker på om den er overflødig. Siden vi vil lage en Oppgave for hver oppdatering av "tilstanden" til hvert enkelt event